From 90d4c44a0dd0d65c6fb3ab2417b83d700c8413ae Mon Sep 17 00:00:00 2001
From: Davide Franco <bacula-dev@dflc.ch>
Date: Mon, 25 Sep 2017 08:58:16 +0200
Subject: [PATCH] Ensure provided period is a valid integer and a known value

  - if provided period is not an integer or a known value, we
    raise an exception
---
 client-report.php | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/client-report.php b/client-report.php
index 0110ecc8..5387011e 100644
--- a/client-report.php
+++ b/client-report.php
@@ -64,6 +64,15 @@
 
        $period = CHttpRequest::get_Value('period');
 
+       // Check if period is an integer and listed in known periods
+       if(!array_key_exists( $period, $periods_list)) {
+          throw new Exception('Critical: provided value for (period) is unknown or not valid');
+       }
+
+       if(!filter_var($period, FILTER_VALIDATE_INT)) {
+          throw new Exception('Critical: provided value for (period) is unknown or not valid');
+       }
+
        $view->assign( 'no_report_options', 'false');
        
        // Client informations
